### PR TITLE
feat: credit card number issuer & formats updated

### PR DESCRIPTION
**Pull request by arulprabakaran**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngneat/falso/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: packages/falso/src/lib/credit-card-number.ts [hunk 1]
```diff
@@ -12,19 +27,38 @@ import { randNumber } from './number';
...
+    issuer?: Issuer;
+  }
+>(options?: Options) {
   return fake(() => {
-    let cardNumber = '';
+    let card = rand(data);
+
+    if (options?.issuer) {
+      const issuerCard = data.find((card) => {
+        return card.issuer === options?.issuer;
```
Comment (NetanelBasal) [inline packages/falso/src/lib/credit-card-number.ts:48]:
```suggestion
        return card.issuer === options.issuer;
```

File: packages/falso/src/lib/credit-card-number.ts [hunk 2]
```diff
@@ -12,19 +27,38 @@ import { randNumber } from './number';
...
-) {
+
+export function randCreditCardNumber<
+  Options extends FakeOptions & {
+    issuer?: Issuer;
+  }
+>(options?: Options) {
   return fake(() => {
-    let cardNumber = '';
+    let card = rand(data);
```
Comment (NetanelBasal) [inline packages/falso/src/lib/credit-card-number.ts:44]: We can skip this function when we have issuer & issuerCard

Comment (arulprabakaran) [reply packages/falso/src/lib/credit-card-number.ts]: Usage of fake function is removed @NetanelBasal

File: packages/falso/src/lib/credit-card-number.ts [hunk 3]
```diff
@@ -12,19 +27,42 @@ import { randNumber } from './number';
...
-
-    return cardNumber;
-  }, options);
+
+export function randCreditCardNumber<
+  Options extends FakeOptions & {
+    issuer?: Issuer;
+  }
+>(options?: Options) {
+  let card = rand(data);
```
Comment (NetanelBasal) [inline packages/falso/src/lib/credit-card-number.ts:43]: I'm not sure why you changed the implementation. Each function should use the `fake` function.

Comment (arulprabakaran) [reply packages/falso/src/lib/credit-card-number.ts:43]: Sorry I misunderstood your previous comment. You are telling to skip `rand` function, right?

Comment (arulprabakaran) [reply packages/falso/src/lib/credit-card-number.ts:43]: Anyways I need to a random value from data to get a format from any one provider, that's why I did that. Do you think we no need to do that?

**Pull request merged**